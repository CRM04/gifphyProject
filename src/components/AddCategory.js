import React, { useState } from 'react';
import propTypes from 'prop-types';


export const AddCategory = ({ setHeroes }) => {
    const [inputValue, setinputValue] = useState('');

    const inputValueChange = (e) => {
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 1) {
            setHeroes(her => [inputValue, ...her]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input type="text" value={inputValue} onChange={inputValueChange} ></input>
        </form>
    );

}

AddCategory.propTypes = {
    setHeroes: propTypes.func.isRequired
}

