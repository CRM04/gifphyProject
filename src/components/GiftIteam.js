import React from 'react';
import propTypes from 'prop-types';

export const GiftItem = ({ title, url }) => {

    return (
        <div className="card">
            <img src={url} alt={title} ></img>
            <p> {title} </p>
        </div>
    );
}

GiftItem.prototypes = {
    title: propTypes.string.isRequired,
    url: propTypes.string.isRequired
}