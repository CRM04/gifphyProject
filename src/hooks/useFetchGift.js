import { useState, useEffect } from 'react';
import { getGifts } from '../helpers/getGifs';

export const useFetchGif = (category) => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifts(category).then(res => {
            console.info(res)
            setstate({
                data: res,
                loading: false
            })
        }).catch(err => {
            console.error(err)
        });
    }, [category]);

    return state;
}