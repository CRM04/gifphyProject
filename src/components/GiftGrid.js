import React from 'react';
import { useFetchGif } from '../hooks/useFetchGift';
import { GiftItem } from './GiftIteam';
import PropTypes from 'prop-types';

export const GiftGrid = ({ heroe }) => {

    const { data, loading } = useFetchGif(heroe);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn"> {heroe} </h3>
            { loading && <p>Loading... </p> }
            <div className="card-grid animate__animated animate__fadeIn">
                {
                    data.map(gif => <GiftItem {...gif} key={gif.id} />)
                }
            </div>
        </>
    );
}

GiftGrid.propTypes = {
    heroe: PropTypes.string.isRequired
}