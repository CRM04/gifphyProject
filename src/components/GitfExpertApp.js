import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GiftGrid } from './GiftGrid';

const GiftExpertApp = ( {categories = []} ) => {

    const [Heroes, setHeroes] = useState(categories);

    return (
        <div>
            <h2> GitfExpertApp </h2>
            <AddCategory setHeroes={setHeroes} />
            <hr></hr>
            {Heroes.map(heroe => <GiftGrid heroe={heroe} key={heroe} />)}
        </div>
    );


}

export default GiftExpertApp;