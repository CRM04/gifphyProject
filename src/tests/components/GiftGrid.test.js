import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GiftGrid } from '../../components/GiftGrid';
import { useFetchGif } from '../../hooks/useFetchGift';

jest.mock('../../hooks/useFetchGift');

describe('GiftGrid test suit', () => {

    test('Giftgrid should match', () => {
        useFetchGif.mockReturnValue({ data: [], loading: true });
        const wrapper = shallow(<GiftGrid heroe={'Naruto'} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should show gifs when using useFetchGifs and hide the loading section', () => {
        const gifs = [{
            id: '1',
            url: 'http://google.com/imagen.jpg',
            title: 'GoogleImage'
        }];

        useFetchGif.mockReturnValue({ data: gifs, loading: false });
        const wrapper = shallow(<GiftGrid heroe="Goku" />);
        expect(wrapper.find('GiftItem').length).toBe(gifs.length);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper).toMatchSnapshot();
    });

});
