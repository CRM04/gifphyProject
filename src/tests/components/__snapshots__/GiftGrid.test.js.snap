// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftGrid test suit Giftgrid should match 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Naruto
     
  </h3>
  <p>
    Loading... 
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;

exports[`GiftGrid test suit Should show gifs when using useFetchGifs and hide the loading section 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GiftItem
      id="1"
      key="1"
      title="GoogleImage"
      url="http://google.com/imagen.jpg"
    />
  </div>
</Fragment>
`;
